core/src/main/java/com/linecorp/armeria/server/DefaultRoutingContext.java

```diff
@@ -125,7 +131,7 @@ public QueryParams params() {
     @Nullable
     @Override
     public MediaType contentType() {
-        return headers.contentType();
+        return contentType;
```

Comment:
Could you also apply the same approach to `headers.method()`?

Reply:
I didn't do that because DefaultRequestHeaders caches the method. But it'd be better to cache the method from here just in case. Will fix it. Thank you!